torch/autograd/gradcheck.py

```diff
@@ -201,6 +203,32 @@ def get_analytical_jacobian(input, output, nondet_tol=0.0, grad_out=1.0):
 
     return jacobian, reentrant, correct_grad_sizes, correct_grad_types
 
+def test_batched_grad(input, output):
+    diff_input_list = list(iter_tensors(input, True))
+    grad = functools.partial(torch.autograd.grad, output, diff_input_list, retain_graph=True, allow_unused=True)
+
+    def vjp(v):
+        results = grad(v)
+        results = tuple(grad if grad is not None else
+                        torch.zeros([], dtype=output.dtype, device=output.device).expand(inp.shape)
+                        for grad, inp in zip(results, diff_input_list))
+        return results
+
+    grad_outputs = [torch.randn_like(output) for _ in range(2)]
+
+    expected = [vjp(gO) for gO in grad_outputs]
+    expected = [torch.stack(shards) for shards in zip(*expected)]
+
+    # Squash warnings since these are expected to happen in most cases
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="Batching rule not implemented")
+        warnings.filterwarnings("ignore", message="torch.vmap is an experimental prototype")
+        result = vmap(vjp)(torch.stack(grad_outputs))
+    for res, exp in zip(result, expected):
+        if torch.allclose(res, exp):
+            continue
+        return fail_test("TODO")
```

Comment:
TODO: a nice error message telling folks what they should do if they see a test failure.

I'm trying to get signal from this PR first though on the CI